lib/src/command/lish.dart

```diff
@@ -251,64 +271,120 @@ the \$PUB_HOSTED_URL environment variable.''',
       usageException('Cannot use both --force and --dry-run.');
     }
 
-    if (entrypoint.root.pubspec.isPrivate) {
+    if (_fromArchive != null && _toArchive != null) {
+      usageException('Cannot use both --from-archive and --to-archive.');
+    }
+
+    if (_fromArchive != null && dryRun) {
+      usageException('Cannot use both --from-archive and --dry-run.');
```

Comment:
We could do the same level of minimal validation, couldn't we?

Reply:
I don't think it really makes sense - we could extract the archive and do the full validation - but I don't think we want